ts/kpt-functions/src/types.ts

```diff
@@ -353,6 +353,9 @@ interface ConfigMap extends KubernetesObject {
   data?: { [key: string]: string };
 }
 
-function isConfigMap(o: any): o is ConfigMap {
+/**
+ * Type guard for ConfigMap.
+ */
+export function isConfigMap(o: any): o is ConfigMap {
```

Comment:
what's the context for needing this? Note that functions implementations have access to ConfigMap class in `gen` that provides such a method. This is for use by the framework library so it doesn't depend on `gen` classes and cause circular dependency.

Reply:
Refactored to use gen, thanks for the pointer about circular dependency!